### ISSUE TITLE
Problem with overriding mochawsome.json

### ISSUE THREAD

**Issue by joszo** [YOU54F/cypress-slack-reporter#400]
Hi,

If we will run e.g `"scripts": {    "slack-report": "./node_modules/.bin/cypress-slack-reporter-full",
} `

Which have: 

 mochawesomeReporterOptions: {
            reportDir: "cypress/reports/mocha",
            quiet: true,
            overwrite: true,
            html: false,
            json: true
        }

Overwrite is set to true, so if user will have more then one test the json file will be always overwritten, and we can't merge couple of reports into one. 

Could you check it?

**Comment by YOU54F** [comment]
The default value is `true` for overwrite

https://github.com/adamgruber/mochawesome-report-generator#options

you will need to manually set it to `false` in your `reporterOpts.json`

**Comment by joszo** [comment]
I've set it to the false:

```{
    "reporterEnabled": "mochawesome",
    "mochawesomeReporterOptions": {
      "reportDir": "cypress/reports/mocha",
      "quiet": false,
      "overwrite": false,
      "html": false,
      "json": true
    }
  }
```

Here you have a number of tests which was performed:
![Screen Shot 2020-02-03 at 9 26 16 AM](https://user-images.githubusercontent.com/32454314/73636933-57647a80-4667-11ea-8eb4-d0636b6ac3c1.png)

And the final json looks like that:

`{
stats: {
suites: 1,
tests: 3,
passes: 3,
pending: 0,
failures: 0,
start: "2020-01-22T11:57:18.156Z",
end: "2020-01-22T11:57:18.156Z",
duration: 715,
testsRegistered: 3,
passPercent: 100,
pendingPercent: 0,
other: 0,
hasOther: false,
skipped: 0,
hasSkipped: false
},
results: [
{
uuid: "43179a5f-35e3-40c5-be22-ee15ab9a3bb2",
title: "",
fullFile: "",
file: "",
beforeHooks: [ ],
afterHooks: [ ],
tests: [ ],
suites: [
{
uuid: "44f4dd07-bf2c-4aa9-a722-e7dc9831564a",
title: "Window",
fullFile: "",
file: "",
beforeHooks: [ ],
afterHooks: [ ],
tests: [
{
title: "cy.window() - get the global window object",
fullTitle: "Window cy.window() - get the global window object",
timedOut: null,
duration: 368,
state: "passed",
speed: "slow",
pass: true,
fail: false,
pending: false,
context: null,
code: "// https://on.cypress.io/window cy.window().should('have.property', 'top');",
err: { },
uuid: "424065d5-b36a-47af-874f-9837519acd3f",
parentUUID: "44f4dd07-bf2c-4aa9-a722-e7dc9831564a",
isHook: false,
skipped: false
},
{
title: "cy.document() - get the document object",
fullTitle: "Window cy.document() - get the document object",
timedOut: null,
duration: 216,
state: "passed",
speed: "slow",
pass: true,
fail: false,
pending: false,
context: null,
code: "// https://on.cypress.io/document cy.document().should('have.property', 'charset').and('eq', 'UTF-8');",
err: { },
uuid: "361d440b-83db-4d0c-94c0-9e17734aa1c0",
parentUUID: "44f4dd07-bf2c-4aa9-a722-e7dc9831564a",
isHook: false,
skipped: false
},
{
title: "cy.title() - get the title",
fullTitle: "Window cy.title() - get the title",
timedOut: null,
duration: 131,
state: "passed",
speed: "slow",
pass: true,
fail: false,
pending: false,
context: null,
code: "// https://on.cypress.io/title cy.title().should('include', 'Kitchen Sink');",
err: { },
uuid: "5418898b-7a46-4c31-8a4d-012f17267c8f",
parentUUID: "44f4dd07-bf2c-4aa9-a722-e7dc9831564a",
isHook: false,
skipped: false
}
],
suites: [ ],
passes: [
"424065d5-b36a-47af-874f-9837519acd3f",
"361d440b-83db-4d0c-94c0-9e17734aa1c0",
"5418898b-7a46-4c31-8a4d-012f17267c8f"
],
failures: [ ],
pending: [ ],
skipped: [ ],
duration: 715,
root: false,
rootEmpty: false,
_timeout: 2000
}
],
passes: [ ],
failures: [ ],
pending: [ ],
skipped: [ ],
duration: 0,
root: true,
rootEmpty: true,
_timeout: 2000
}
],
meta: {
mocha: {
version: "5.2.0"
},
mochawesome: {
options: {
quiet: true,
reportFilename: "mochawesome",
saveHtml: false,
saveJson: true,
consoleReporter: "spec",
useInlineDiffs: false
},
version: "4.1.0"
},
marge: {
options: {
id: "default",
reportDir: "cypress/reports/mocha",
quiet: true,
overwrite: true,
html: false,
json: true
},
version: "4.0.1"
}
}
}`

There are only 3 tests instead of 97..., I'm not sure if it is my configuration problem, or it doesn't work correctly with more then one spec file.

**Comment by YOU54F** [comment]
yeah i do need to update the docs, mocha-multi-reporters isn't maintained and cypress-multi-reporters will work with all versions of mocha.

can you provide an example project? I have working examples here that generate multiple spec files per test and use mochawesome-merge to combine them into a single json file

https://github.com/YOU54F/cypress-docker-typescript/tree/master/e2e
https://github.com/YOU54F/cypress-docker/tree/master/e2e

**Comment by joszo** [comment]
I will check your examples, maybe I would find some solution to my problem.

In the meantime the sample project where I tried to integrate cypress+slack_reporter+circle_ci is here: https://github.com/joszo/circle-ci-cypress

**Comment by YOU54F** [comment]
Does that project run in ci?

How do you have an executable here

./node_modules/.bin/cypress-slack-reporter-full

The executable Is called cypress-slack-reporter and your circleci yml doesn’t appear to have anything that runs a test?

**Comment by YOU54F** [comment]
@joszo should be released shortly in https://www.npmjs.com/package/cypress-slack-reporter/v/0.5.1

see branch 

https://github.com/YOU54F/cypress-slack-reporter/pull/422/files

**Comment by joszo** [comment]
Great to here it, after release I will test it, and inform you if everything works correctly :)

**Comment by YOU54F** [comment]
I've worked out two additional issues

1. after merging the report, the script would leave the unmerged json files. The resulted in an error, because the slack reporter is only expecting 1 json file. This will be resolved in 0.5.1 by deleting the additional json files (and will only work with mochawesome-merge v2.0.1 or lower)
2. mochawesome 4.0 drops the reportDir option and replaces with a files option with a globby pattern. this will be released in 0.6.0

thanks for your report @joszo I have taken a fork of your repo, so I will release and test out and get back to you :)

**Comment by joszo** [comment]
Hello, 

I've tested on Friday your fix and right now html report is generated from the multiple JSON files. 

As you mentioned it is important to use it with **mochawesome-merge v2.0.1 or lower**. 

And using `"scripts": { "slack-report": "./node_modules/.bin/cypress-slack-reporter-full", }` with reporterOpts.json doesn't save reports in desired folder (I think reportedOpts doesn't wokr at all if we want to overrider configuration used in cypress-slack-reporter).  

Example: 
`      "reportDir": "cypress/reports/mochawesome-report",` 

will save our reports in `            reportDir: "cypress/reports/mocha",`

**Comment by YOU54F** [comment]
yeah the scripted runner won't honour reporterOpts's as it sets it own.

https://github.com/YOU54F/cypress-slack-reporter#scripted-runner

it was only ever meant to be used as a example. but it can probably be setup so that it will consume the reporterOpts at the root of a project.

**Comment by YOU54F** [comment]
Is this issue now resolved for you, or is there additional work required?

**Comment by AishwaryaRavindra** [comment]
I am facing the Mochawsome merge html report , i am getting the Html report, but it is generating separated in each test, even though if  i have added the Config also, not able to generating the report in one single file.  below is the snap.

![Screenshot (22)](https://github.com/YOU54F/cypress-plugins/assets/114048755/618eb132-d07d-4f62-a4f8-d6a8083a018f)

**Comment by YOU54F** [comment]
Please create a reproducible example.

note mocha-awesome merge is not part of this project.

als you command for `combine-reports` looks incorrect, you are missing a space in the command

**Comment by AishwaryaRavindra** [comment]
I am getting your point, if possible can you provide me the Proper Config to add in JSON file, for mocha reporting, and Merge into a single Html file.

**Comment by YOU54F** [comment]
will all due respect, no, my working is shown working here and tested in CI, and as said, you should read the documentation on the site of the tool you are using
